### PR TITLE
doc: fixes

### PR DESCRIPTION
**Pull request by ayman-alkom**
- added some clarifications to the error renderer setting with 3.3+

-  replace depricated $event->subject() with $event->getSubject()

### CODE REVIEW

File: docs/listeners/jsonapi.rst [hunk 1]
```diff
@@ -110,8 +110,37 @@ Exception handler
...
+  class AppController extends Controller {
+
+    public function initialize()
+    {
+      parent::initialize();
+      $this->Crud->config(['listeners.api.exceptionRenderer' => 'App\Error\JsonApiExceptionRenderer']);
+    }
+  }
+
+``Note:`` however if you are using CakePHP 3.3's PSR7 middleware feature the ``exceptionRenderer``
```
Comment (davidyell) [inline docs/listeners/jsonapi.rst]: I would prefer a capital letter here still.

File: docs/listeners/api.rst [hunk 2]
```diff
@@ -115,9 +115,19 @@ class and enabling it with the `exceptionRenderer` configuration option.
     }
   }
 
-However if you are using CakePHP 3.3's PSR7 middleware feature the ``exceptionRenderer``
+``Note:`` however if you are using CakePHP 3.3's PSR7 middleware feature the ``exceptionRenderer``
```
Comment (ADmad) [inline docs/listeners/api.rst]: Using `**Note:**` makes lot more sense here.  Also "3.3's" should be changed to "3.3+'s".

File: docs/listeners/jsonapi.rst (hunk 1 above)
Comment (ADmad) [inline docs/listeners/jsonapi.rst]: Same comment as i made above.

**Comment by ayman-alkom** [comment]
hi @davidyell hi @ADmad I hope the changes made as described

File: docs/listeners/jsonapi.rst [hunk 3]
```diff
@@ -110,8 +110,37 @@ Exception handler
 -----------------
 
 The JsonApi listener overrides the ``Exception.renderer`` for ``jsonapi`` requests,
-so in case of an error, a standardized error in JSON API format will be returned
-for both errors/exceptions and validation errors.
+so in case of an error, a standardized error will be returned, in either
+``json`` or ``xml`` - according to the API request type.
+
+Create a custom exception renderer by extending the Crud JsonApiExceptionRenderer
+class and enabling it with the `exceptionRenderer` configuration option.
```
Comment (ADmad) [inline docs/listeners/jsonapi.rst]: `exceptionRenderer` should be wrapped in double tildas.

**Pull request merged**